client/allocrunner/taskrunner/task_runner.go

```diff
@@ -1041,10 +1041,11 @@ func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig {
 	if alloc.AllocatedResources != nil && len(alloc.AllocatedResources.Shared.Networks) > 0 {
 		allocDNS := alloc.AllocatedResources.Shared.Networks[0].DNS
 		if allocDNS != nil {
+		interpolatedNetworks := taskenv.InterpolateNetworks(env, alloc.AllocatedResources.Shared.Networks)
```

Comment:
My suspicion is that there should be a better place to put this (ie this should probably be in a hook file), but this does have the value of actually working